test-cypress/support/repository-commands.js

```diff
@@ -17,3 +20,66 @@ Cypress.Commands.add('deleteRepository', (id) => {
     const url = REPOSITORIES_URL + id;
     cy.request('DELETE', url).should((response) => expect(response.status).to.equal(200));
 });
+
+Cypress.Commands.add('importRDF', (repositoryId, rdf) => {
+    var name = "Text snippet";
+    const importData = {
+     "name": name,
+       "status":"NONE",
+       "message":"",
+       "context":"",
+       "type":"text",
+       "format":"text/turtle",
+       "data":"@base <http://example.org/> .\n@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .\n@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .\n@prefix foaf: <http://xmlns.com/foaf/0.1/> .\n@prefix rel: <http://www.perceive.net/schemas/relationship/> .\n\n<#green-goblin>\n    rel:enemyOf <#spiderman> ;\n    a foaf:Person ;    # in the context of the Marvel universe\n    foaf:name \"Green Goblin\" .\n\n<#spiderman>\n    rel:enemyOf <#green-goblin> ;\n    a foaf:Person ;\n    foaf:name \"Spiderman\", \"Человек-паук\"@ru .",
+    }
+    cy.request({
+        method: 'POST',
+        url: IMPORT_URL + repositoryId + '/text',
+        body: importData,
+        headers: {
+            'X-GraphDB-Repository': repositoryId,
+        }
+    }).should((response) => expect(response.status).to.equal(202));
+    waitImport(repositoryId, name);
+});
+
+let waitImport = function(repositoryId, name) {
+    cy.wait(1000);
+    cy.request({
+        method: 'GET',
+        url: IMPORT_URL + repositoryId,
+        headers: {
+            'X-GraphDB-Repository': repositoryId,
+        }
+    }).then((response) => {
+        if (response.status == 200 && Cypress._.find(response.body, function(o) {return o.name === name}).status == 'DONE') return;
+        waitImport(repositoryId, name);
```

Comment:
I tried with the request timeout, but it did not stop the recursion, so I added a pollTimeout parameter in the recursion to track something like a total timeout. It is not the perfectly exact time, since on each call I decrease it with the wait time + response time, but this should be close enough.